internal/extramanifest/extramanifest.go

```diff
@@ -179,79 +192,149 @@ func ApplyExtraManifest(ctx context.Context, dc dynamic.Interface, restMapper me
 	}
 
-	// MachineConfig can trigger reboot and this will result in not meeting KPI downtime requirement
-	// so raise validation error if there is a MachineConfig in extramanifests
-	for _, manifest := range manifests {
-		if manifest.GetKind() == "MachineConfig" && manifest.GetAPIVersion() == "machineconfiguration.openshift.io/v1" {
-			return NewEMFailedValidationError("Using MachingConfigs in extramanifests is not allowed")
+	sortedManifests, err := common.SortAndGroupByApplyWave[*unstructured.Unstructured](manifests)
+	if err != nil {
+		return validationWarn, fmt.Errorf("failed to sort and group manifests: %w", err)
+	}
+
+	// A namespace set saves all created namespaces for dryrun
+	fakedNamespacesForDryrun := make(map[string]bool)
+	// Cleanup any faked namespaces at the end of validation
+	defer func() error {
+		for name := range fakedNamespacesForDryrun {
+			ns := &corev1.Namespace{}
+			ns.SetName(name)
+			if err := h.Client.Delete(ctx, ns); err != nil {
+				if !k8serrors.IsNotFound(err) {
+					return fmt.Errorf("failed to delete namespace %s: %w", ns.Name, err)
+				}
+			}
+			h.Log.Info(fmt.Sprintf("Deleted faked namespace %s for dryrun", ns.Name))
 		}
-	}
+		return nil
+	}()
+
+	dryRun := true                          // Apply manifest with dryRun
+	nsManifestsSet := make(map[string]bool) // A namespace set collects appeared namespaces in configmaps
+
+	for _, manifestGroup := range sortedManifests {
+		for _, manifest := range manifestGroup {
+			mKind := manifest.GetKind()
+			mApiVersion := manifest.GetAPIVersion()
+			mName := manifest.GetName()
+			mNamespace := manifest.GetNamespace()
+
+			// Verify resource
+			if mName == "" {
+				errs = append(errs, fmt.Sprintf("%s resource name is empty", mKind))
+				continue
+			}
 
-	// a map to track all the unknown CRDs in one shot
-	unknownCRDs := make(map[string]string)
-
-	// Apply manifest with dryRun
-	dryRun := true
-
-	// Collect unknownCRDs errors and continue to treat them as part of warning.
-	// Any other error should be treated as real error and return early (which should be priority over warnings)
-	for _, manifest := range manifests {
-		if err := ApplyExtraManifest(ctx, h.DynamicClient, h.Client.RESTMapper(), manifest, dryRun); err != nil {
-			defaultErrMsg := fmt.Sprintf("failed to apply manifest with dryrun: %v", err.Error())
-			var groupDiscoveryErr *discovery.ErrGroupDiscoveryFailed
-			switch {
-			case errors.As(err, &groupDiscoveryErr) || meta.IsNoMatchError(err): // Capture unknown CRD issue
-				key := manifest.GroupVersionKind().String()
-				if _, exists := unknownCRDs[key]; !exists {
-					unknownCRDs[key] = manifest.GetName()
-					h.Log.Info("unknown CRD", "CRD", key)
+			// Verify disallowed resource
+			if kinds, exists := unSupportedResources[mApiVersion]; exists {
+				if lo.Contains(kinds, mKind) {
+					errs = append(errs, fmt.Sprintf("%s[%s] in extramanifests is not allowed", mKind, mName))
+					continue
```

Comment:
Ah nice..so we now collecting all the errors before returning yeah? (unlike prev we were returning early?)

Reply:
Exactly, collect all errors and warnings.